src/incubator/Dialog/index.tsx

```diff
@@ -202,8 +213,7 @@ const Dialog = (props: DialogProps, ref: ForwardedRef<DialogImperativeMethods>)
     <GestureDetector gesture={panGesture}>
       {/* @ts-expect-error should be fixed in version 3.5 (https://github.com/software-mansion/react-native-reanimated/pull/4881) */}
       <View {...containerProps} reanimated style={style} onLayout={onLayout} ref={setRef} testID={testID}>
-        {headerProps && <DialogHeader {...headerProps}/>}
-        {children}
+        <DialogContent headerProps={headerProps} children={children}/>
```

Comment:
This is confusing... You're passing 'headerProps' as a prop to the DialogContent that you got from the hook that sets 'headerProps' in the 'DialogContent' it returns... Why not just pass it to the hook with the other props and set it there, only rendering the returned component in the Dialog?
Same as 'children'...